osquery/events/windows/windowseventlogpublisher.cpp

```diff
@@ -0,0 +1,205 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed in accordance with the terms specified in
+ *  the LICENSE file found in the root directory of this source tree.
+ */
+
+#include <osquery/config/config.h>
+#include <osquery/events/windows/windowseventlogpublisher.h>
+#include <osquery/flags.h>
+#include <osquery/logger.h>
+#include <osquery/registry_factory.h>
+
+#include <plugins/config/parsers/feature_vectors.h>
+
+namespace pt = boost::property_tree;
+
+namespace osquery {
+namespace {
+const int kCharFreqVectorLen{256};
+
+Status loadCharacterFrequencyMap(std::vector<double>& character_frequency_map) {
```

Comment:
I don't have a better solution here, and this seems to make sense, but the placement of this parser feels out of place to me just a bit. I don't have any good suggestions on where it might better be housed, but curious if we explored any other options?

Reply:
This has been moved away from the powershell_events implementation file into the publisher, which is the same scope where it is now used (so that we reduce the scope as much as possible). 

In the future, if cosineSimilarity becomes useful for other applications, it would be nice to migrate this into the configuration + utility classes